tests/migration_test.go

```diff
@@ -3798,6 +3799,62 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 		By("Waiting for VMI to disappear")
 		tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
 	})
+
+	Context("with a live-migration in flight", func() {
+		It("there should always be a single active migration per VMI", func() {
+			By("Starting a VMI")
+			vmi := libvmi.NewCirros(
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			)
+			vmi = runVMIAndExpectLaunch(vmi, 240)
+
+			By("Starting a Migration")
+			migration := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+			migration = tests.RunMigration(virtClient, migration)
+			createdMigrations := []*v1.VirtualMachineInstanceMigration{migration}
+
+			By("Checking that there always is at most one migration running")
+			Consistently(func() int {
+				concurrentVMIM := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+				// not checking err as the migration creation will be blocked immediately by virt-api's validating webhook
+				// if another one is currently running
+				concurrentVMIM, err = virtClient.VirtualMachineInstanceMigration(concurrentVMIM.Namespace).Create(concurrentVMIM, &metav1.CreateOptions{})
+				if err == nil {
+					createdMigrations = append(createdMigrations, concurrentVMIM)
+				}
+
+				labelSelector, err := labels.Parse(fmt.Sprintf("%s in (%s)", v1.MigrationSelectorLabel, vmi.Name))
+				Expect(err).ToNot(HaveOccurred())
+				listOptions := &metav1.ListOptions{
+					LabelSelector: labelSelector.String(),
+				}
+				migrations, err := virtClient.VirtualMachineInstanceMigration(concurrentVMIM.Namespace).List(listOptions)
+				Expect(err).ToNot(HaveOccurred())
+
+				activeMigrations := 0
+				for _, vmim := range migrations.Items {
+					switch vmim.Status.Phase {
+					case v1.MigrationScheduled, v1.MigrationPreparingTarget, v1.MigrationTargetReady, v1.MigrationRunning:
+						activeMigrations += 1
+					}
+				}
+				return activeMigrations
+
+			}, time.Second*30, time.Second*1).Should(BeNumerically("<=", 1))
+
+			By("Deleting all created migrations")
```

Comment:
You don't need to clean migrations and vmis.

Reply:
Why not? the test just before this one does it as well, otherwise, they will be cleaned up only after all tests have been run(?).

Reply:
I think it is a historical thing and you can make the cleaning easier. `tests.BeforeTestCleanup()` should clean them.